TezosKit/Client/AbstractClient.swift

```diff
@@ -0,0 +1,80 @@
+// Copyright Keefer Taylor, 2019
+
+import Foundation
+
+/// An abstract network client that can send RPC requests.
+public class AbstractClient {
+
+  /// The URL session that will be used to manage URL requests.
+  private let urlSession: URLSession
+
+  /// A URL pointing to a remote node that will handle requests made by this client.
+  private let remoteNodeURL: URL
+
+  /// A response handler for RPCs.
+  private let responseHandler: RPCResponseHandler
+
+  /// The queue that callbacks from requests will be made on.
+  private let callbackQueue: DispatchQueue
+
+  /**
+   * Initialize a new AbstractNetworkClient.
+   *
+   * - Parameter remoteNodeURL: The path to the remote node.
+   * - Parameter urlSession: The URLSession that will manage network requests.
+   * - Parameter callbackQueue: A dispatch queue that callbacks will be made on.
+   * - Parameter responseHandler: An object which will handle responses.
+   */
+  public init(
+    remoteNodeURL: URL,
+    urlSession: URLSession,
+    callbackQueue: DispatchQueue,
+    responseHandler: RPCResponseHandler
+  ) {
+    self.remoteNodeURL = remoteNodeURL
+    self.urlSession = urlSession
+    self.callbackQueue = callbackQueue
+    self.responseHandler = responseHandler
+  }
+
+  /**
+   * Send an RPC as a GET or POST request.
+   */
+  public func send<T>(rpc: RPC<T>) {
+    guard let remoteNodeEndpoint = URL(string: rpc.endpoint, relativeTo: self.remoteNodeURL) else {
+      let error = TezosKitError(kind: .unknown, underlyingError: nil)
```

Comment:
Would be nice if this explained that it was an invalid URL error and had the endpoint and remotenodeurl in the description